tests/system/test_transfer_manager.py

```diff
@@ -75,10 +92,100 @@ def test_upload_many_skip_if_exists(
 
 def test_download_many(listable_bucket):
     blobs = list(listable_bucket.list_blobs())
-    tempfiles = [tempfile.TemporaryFile(), tempfile.TemporaryFile()]
-    BLOB_FILE_PAIRS = zip(blobs[:2], tempfiles)
-
-    results = transfer_manager.download_many(BLOB_FILE_PAIRS)
-    assert results == [None, None]
-    for fp in tempfiles:
-        assert fp.tell() != 0
+    with tempfile.TemporaryDirectory() as tempdir:
+        filenames = [
+            os.path.join(tempdir, "file_a.txt"),
+            os.path.join(tempdir, "file_b.txt"),
+        ]
+        BLOB_FILE_PAIRS = zip(blobs[:2], filenames)
+
+        results = transfer_manager.download_many(
+            BLOB_FILE_PAIRS,
+            worker_type=transfer_manager.PROCESS,
+            deadline=DEADLINE,
+        )
+        assert results == [None, None]
+        for count, filename in enumerate(filenames):
+            with open(filename, "rb") as fp:
+                assert len(fp.read()) == blobs[count].size
+
+
+def test_download_many_with_threads_and_file_objs(listable_bucket):
+    blobs = list(listable_bucket.list_blobs())
+    with tempfile.TemporaryFile() as file_a, tempfile.TemporaryFile() as file_b:
+        tempfiles = [file_a, file_b]
+        BLOB_FILE_PAIRS = zip(blobs[:2], tempfiles)
+
+        results = transfer_manager.download_many(
+            BLOB_FILE_PAIRS,
+            worker_type=transfer_manager.THREAD,
+            deadline=DEADLINE,
+        )
+        assert results == [None, None]
+        for fp in tempfiles:
+            assert fp.tell() != 0
+
+
+def test_download_chunks_concurrently(shared_bucket, file_data):
+    # Upload a big file
+    source_file = file_data["big"]
+    upload_blob = shared_bucket.blob("chunky_file")
+    upload_blob.upload_from_filename(source_file["path"])
+    upload_blob.reload()
+    size = upload_blob.size
+    chunk_size = size // 32
+    midpoint = size // 2
+
+    # Get a fresh blob obj w/o metadata for testing purposes
+    download_blob = shared_bucket.blob("chunky_file")
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        full_filename = os.path.join(tempdir, "chunky_file")
+        transfer_manager.download_chunks_concurrently(
+            download_blob,
+            full_filename,
+            chunk_size=chunk_size,
+            deadline=DEADLINE,
+        )
+        with open(full_filename, "rb") as file_obj:
+            assert _base64_md5hash(file_obj) == source_file["hash"]
+
+        # Now test for case where last chunk is exactly 1 byte.
```

Comment:
Thanks for the thorough tests! I especially like this one; really helped check the cursor moving through the last byte.